lib/modules/manager/maven-wrapper/artifacts.spec.ts

```diff
@@ -0,0 +1,211 @@
+          path: 'maven.mvn/wrapper/maven-wrapper.properties',
+          type: 'addition',
+        },
+      },
+    ];
+    expect(execSnapshots[2].cmd).toContain('java 8.0.1');
+    expect(updatedDeps).toEqual(expected);
+  });
+
+  it('Should update when it is maven wrapper', async () => {
+    mockMavenFileChangedInGit();
+    mockExecAll({ stdout: '', stderr: '' });
+    const updatedDeps = await updateArtifacts({
+      packageFileName: 'maven',
+      newPackageFileContent: '',
+      updatedDeps: [{ depName: 'org.apache.maven.wrapper:maven-wrapper' }],
+      config: { currentValue: '3.3.1', newValue: '3.3.1' },
+    });
+
+    const expected = [
+      {
+        file: {
+          contents: undefined,
+          path: 'maven.mvn/wrapper/maven-wrapper.properties',
+          type: 'addition',
+        },
+      },
+    ];
+    expect(updatedDeps).toEqual(expected);
+  });
+
+  it('Should not update deps when maven-wrapper.properties is not in git change', async () => {
+    mockMavenFileChangedInGit('not-maven-wrapper.properties');
+    mockExecAll({ stdout: '', stderr: '' });
+    const updatedDeps = await updateArtifacts({
+      packageFileName: 'maven',
+      newPackageFileContent: '',
+      updatedDeps: [{ depName: 'org.apache.maven.wrapper:maven-wrapper' }],
+      config: { newValue: '3.3.1' },
+    });
+    expect(updatedDeps).toEqual([]);
+  });
+
+  it('updates with docker', async () => {
+    mockMavenFileChangedInGit();
+    GlobalConfig.set({ localDir: './', binarySource: 'docker' });
+    const execSnapshots = mockExecAll({ stdout: '', stderr: '' });
+    const result = await updateArtifacts({
+      packageFileName: 'maven',
+      newPackageFileContent: '',
+      updatedDeps: [{ depName: 'org.apache.maven.wrapper:maven-wrapper' }],
+      config: { currentValue: '3.3.0', newValue: '3.3.1' },
+    });
+    expect(result).toEqual([
+      {
+        file: {
+          contents: undefined,
+          path: 'maven.mvn/wrapper/maven-wrapper.properties',
+          type: 'addition',
+        },
+      },
+    ]);
+    expect(execSnapshots).toMatchObject([
+      {
+        cmd: 'docker pull renovate/sidecar',
+        options: { encoding: 'utf-8' },
+      },
+      {
+        cmd: 'docker ps --filter name=renovate_sidecar -aq',
+      },
+      {
+        cmd: 'docker run --rm --name=renovate_sidecar --label=renovate_child -v "./":"./" -e BUILDPACK_CACHE_DIR -e CONTAINERBASE_CACHE_DIR -w "./" renovate/sidecar bash -l -c "install-tool java 17.0.0 && ./mvnw wrapper:wrapper"',
+        options: {
+          cwd: './',
+          encoding: 'utf-8',
+          env: {
+            HOME: '/home/user',
+            HTTPS_PROXY: 'https://example.com',
+            HTTP_PROXY: 'http://example.com',
+            LANG: 'en_US.UTF-8',
+            LC_ALL: 'en_US',
+            NO_PROXY: 'localhost',
+            PATH: '/tmp/path',
+          },
+          maxBuffer: 10485760,
+          timeout: 900000,
+        },
+      },
+    ]);
```

Comment:
please format like this:
https://github.com/renovatebot/renovate/blob/07460542850cc79e934a8d4606371c88776e4c0c/lib/modules/manager/gradle-wrapper/artifacts.spec.ts#L201-L221

please also add `binarySource: 'install'` test